src/compiler/sys.ts

```diff
@@ -1256,7 +1256,8 @@ namespace ts {
                 require: (baseDir, moduleName) => {
                     try {
                         const modulePath = resolveJSModule(moduleName, baseDir, nodeSystem);
-                        return { module: require(modulePath), modulePath, error: undefined };
+                        // eslint-disable-next-line no-eval
+                        return { module: eval("module.require")(modulePath), modulePath, error: undefined };

```

Comment:
There's a **much** cleaner way to allow dynamic require in Node, but still be bundled without warnings/errors:
```js
// add this above the call to createSystemWatchFunctions
function moduleRequire(baseModule, request) {
    return baseModule.require(request)
}

// then use it
return { module: moduleRequire(module, modulePath), modulePath: modulePath, error: undefined };
```
This way, webpack doesn't try and follow the require call (so it doesn't warn about it), and everything still works in Node without an `eval` (evil) call.